File: wordteacher/src/main/java/com/example/alexeyglushkov/wordteacher/main/ContextModule.java
package com.example.alexeyglushkov.wordteacher.main;

import android.content.Context;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ContextModule {
    private Context context;

    ContextModule(Context context) {
        this.context = context;
    }

    @Provides @Named("context")
    Context context() {
        return context;
    }

    @Provides @Named("appContext")
    Context applicationContext() {
        return context.getApplicationContext();
    }
}

File: wordteacher/src/main/java/com/example/alexeyglushkov/wordteacher/main/ContextModule.kt
package com.example.alexeyglushkov.wordteacher.main

import android.content.Context

import javax.inject.Named
import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class ContextModule internal constructor(private val context: Context) {

    @Provides
    @Named("context")
    internal fun context(): Context {
        return context
    }

    @Provides
    @Named("appContext")
    internal fun applicationContext(): Context {
        return context.applicationContext
    }
}
